### PR TITLE
V2: Listening to events

### LINKED ISSUE #157: [TASK] Shell Manager - Listen for blueprints
### Overview

The shell manager once it starts, it should query the node to get all the blueprints that this operator is registered in, then it listens for a stream of finality notifications.

In other words, we do this at startup, to get a full view of the world (i.e sync):

```ts
let servicesWithBlueprint = await api.rpc.services.queryServicesWithBlueprintsByOperator(ALICE);
```
> See: https://github.com/webb-tools/tangle/pull/647

Then we should listen for events that are produced by the network, and here are some of them:

```rs
/// An new operator has been registered.
Registered {
  /// The account that registered as a operator.
  provider: T::AccountId,
  /// The ID of the service blueprint.
  blueprint_id: u64,
  /// The preferences for the operator for this specific blueprint.
  preferences: OperatorPreferences,
  /// The arguments used for registration.
  registration_args: Vec<Field<T::Constraints, T::AccountId>>,
}
```
```rs
/// An operator has been unregistered.
Unregistered {
  /// The account that unregistered as am operator.
  operator: T::AccountId,
  /// The ID of the service blueprint.
  blueprint_id: u64,
}
```
These events should tell you that if the current operator has registered or unregistered from a blueprint.
More could be found here: https://github.com/webb-tools/tangle/blob/74ef976802a68b960677277786b91bca13e260a3/pallets/services/src/lib.rs#L196-L330

### Task Checklist

- [x] #178
- [ ] #179
- [ ] #180

### PR DESCRIPTION
**Pull request by tbraun96**
This PR will close #157, #158, #159, #179, #185:

- [ ] https://github.com/webb-tools/gadget/issues/157
- [ ] https://github.com/webb-tools/gadget/issues/158
- [ ] https://github.com/webb-tools/gadget/issues/159
- [ ] https://github.com/webb-tools/gadget/issues/179
- [ ] https://github.com/webb-tools/gadget/issues/185

### CODE REVIEW

File: environments/tangle/src/api.rs [hunk 1]
```diff
@@ -1,131 +1,72 @@
...
+                ret.push(blueprint.1);
+            } else {
+                self.logger.error(format!(
+                    "Failed to fetch blueprint for ID: {}",
+                    blueprint_id
+                ));
+            }
+        }
+
+        Ok(ret)
```
Comment (shekohex) [inline environments/tangle/src/api.rs:70]: This implementation does not seem to be correct. It queries the user services', not the operator's. We should rename this method to be more consistent `services_with_blueprints_by_operator` and use the RPC method provided for you.
https://github.com/webb-tools/tangle/blob/5384f0d19664ab7332a4c191e745c326f0416218/tangle-subxt/src/tangle_testnet_runtime.rs#L686

File: environments/utils/src/transaction_manager/tangle/mod.rs [hunk 2]
```diff
@@ -14,19 +10,11 @@ use std::fmt::Debug;
 #[async_trait]
 #[auto_impl(Arc)]
 pub trait TanglePalletSubmitter: Send + Sync + Debug + 'static {
-    async fn submit_job_result(
+    async fn submit_service_result(
```
Comment (shekohex) [inline environments/utils/src/transaction_manager/tangle/mod.rs:13]:
```suggestion
    async fn submit_job_call_result(
```
is a better name.

**Review by shekohex — COMMENTED** [review]
I know this a WIP, but I took a moment to give you a quick feedback, looks good overall. Looking forward to the final review.

**Comment by drewstone** [comment]
CI is still erroring from outdated v1 logic, what's the status of this @tbraun96 ?

**Comment by tbraun96** [comment]
Some cleanup and fixes in the gadget-sdk remain.

The `gadget-test-utils` crate is obsolete, so, we will likely in another PR clean that up once we figure out a way forward with @shekohex. 

The hierarchy of `gadget-core` -> `gadget-common` -> `shell-manager` is compiling with all the necessary logic in place, which is the big idea behind this composite PR.

We will likely need a new sub-checklist, if we haven't already, to basically (open for discussion):

* Overhaul `gadget-test-utils` (maybe even a rename to `blueprint-test-utils`)
* Add a `BlueprintTestExternalities` that internally uses `MultithreadedTestExternalities` and uses a TBD client
* Add integration tests proving everything works

cc. @drewstone @shekohex

**Comment by drewstone** [comment]
@tbraun96 are all the tasks in this PR complete? Can you check the ones that are off so we can track this properly?

**Pull request merged**